### DISCUSSION TITLE
How to search through entire database in SQLite Studio?

### DISCUSSION THREAD

**Discussion by Valour549** [pawelsalawa/sqlitestudio#4787] [Q&A]
The "filter by name" search bar only gives results from Tables and Columns, is there any way for an advanced search that goes through the entire database for a corresponding string? I'm aware it could take a long time. Thanks!

**Comment by pawelsalawa** [comment]
Create custom SQL function, choose Tcl as implamentation language and use this function: https://github.com/pawelsalawa/sqlitestudio/wiki/Tcl_script_find

**Reply by Valour549** [reply]
Thank you. Is there any way to do wildcard searches like * my_custom_ID  or my_custom_ID * ?

**Reply by pawelsalawa** [reply]
You have to modify the implementation code of the Tcl function. You have to change line:
```
set value [string map [list "'" "''"] [lindex $argv 0]]
```
so it results on list of values passed from arguments, not just single value.

Then you need to change
```
lappend cols "\[$colName\] = '$value'"
```
so it produces SQL condition as you wish, according to multiple values and/or wildcards.
